src/openvic-simulation/modifier/ModifierManager.hpp

```diff
@@ -18,127 +15,123 @@ namespace OpenVic {
 		friend struct RebelManager;
 		friend struct PopManager;
 		friend struct TechnologyManager;
+		friend struct TerrainTypeManager;
 
-		using effect_variant_map_t = ordered_map<Modifier::modifier_type_t, ModifierEffect const*>;
+		using modifier_effect_registry_t = CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque>;
 
 		/* Some ModifierEffects are generated mid-load, such as max/min count modifiers for each building, so
 		 * we can't lock it until loading is over. This means we can't rely on locking for pointer stability,
 		 * so instead we store the effects in a deque which doesn't invalidate pointers on insert.
 		 */
 	private:
-		CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> IDENTIFIER_REGISTRY(modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(leader_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(unit_terrain_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(shared_tech_country_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(technology_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(base_country_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(base_province_modifier_effect);
+		modifier_effect_registry_t IDENTIFIER_REGISTRY(terrain_modifier_effect);
 		case_insensitive_string_set_t complex_modifiers;
-		string_map_t<effect_variant_map_t> modifier_effect_variants;
 
 		IdentifierRegistry<IconModifier> IDENTIFIER_REGISTRY(event_modifier);
-		IdentifierRegistry<Modifier> IDENTIFIER_REGISTRY(static_modifier);
 		IdentifierRegistry<TriggeredModifier> IDENTIFIER_REGISTRY(triggered_modifier);
 
 		ModifierEffectCache PROPERTY(modifier_effect_cache);
 		StaticModifierCache PROPERTY(static_modifier_cache);
-
-		/* effect_validator takes in ModifierEffect const& */
-		NodeTools::key_value_callback_t _modifier_effect_callback(
-			ModifierValue& modifier, Modifier::modifier_type_t type, NodeTools::key_value_callback_t default_callback,
-			ModifierEffectValidator auto effect_validator
-		) const;
-
-	public:
-		bool add_modifier_effect(
-			ModifierEffect const*& effect_cache,
-			std::string_view identifier,
-			bool positive_good,
-			ModifierEffect::format_t format,
+	
+		bool _register_modifier_effect(
+			CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque>& registry,
```

Comment:
Use the `modifier_effect_registry_t` type alias here and everywhere else the registry type is used